src/prompt_systematic_review/role_prompting.py

```diff
@@ -0,0 +1,210 @@
+from prompt_systematic_review.load_hf_data import load_hf_dataset
+import openai
+from typing import List
+import re
+import time
+import json
+from datetime import datetime
+import concurrent.futures
+
+def query_model_with_timeout(
+    prompt: str, question: str, model_name: str, output_tokens: int
+) -> dict:
+    """
+    Query the OpenAI API with a prompt and a question.
+    :param prompt: The prompt to use.
+    :param question: The question to use from the dataset.
+    :param model_name: The OpenAI model to use.
+    :param output_tokens: The maximum number of output tokens to generate.
+    :return: The response from the API.
+    """
+    try:
+        response = openai.chat.completions.create(
+            model=model_name,
+            messages=[
+                {"role": "system", "content": prompt},
+                {"role": "user", "content": question},
+            ],
+            max_tokens=output_tokens,
+        )
+        return response
+    except Exception as e:
+        print(f"An error occurred during API call: {e}")
+        return None
+
+
+def query_model(
+    prompt: str,
+    question: str,
+    model_name: str,
+    output_tokens: int = 300,
+    timeout: float = 15.0,
+) -> dict:
+    """
+    Query the OpenAI API with a timeout.
+    :param prompt: The prompt to use.
+    :param question: The question to use from the dataset.
+    :param model_name: The OpenAI model to use.
+    :param output_tokens: The maximum number of output tokens to generate.
+    :param timeout: Timeout for the request in seconds.
+    :return: The response from the API or None if timeout occurs.
+    """
+    with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
why

Reply:
This is so that if a single request is taking over a certain amount of time, I can move on to the rest of them without getting hung up on that one. Do you suggest I remove it? I have been running into an issue that seems to happen randomly where I run a test that sends a call to the API and it doesn't return, but when I rerun the same test after, seems to work reliably.

Reply:
I have also tested it and doesn't seem to mess with any normal functionality.